src/commands/slow-mode.ts

```diff
@@ -0,0 +1,579 @@
+	.addSubcommandGroup(group => {
+		group.setName('enable')
+			.setDescription('Enable slow mode for a channel')
+			.addSubcommand(cmd => {
+				cmd.setName('auto')
+					.setDescription('Enable auto slow mode')
+					.addChannelOption(option => {
+						option.setName('channel')
+							.setDescription('The channel to enable auto slow mode for');
+						return option;
+					})
+					.addNumberOption(option => {
+						option.setName('window')
+							.setDescription('The time window to use for calculating message rate')
+							.setMinValue(10000);
+						return option;
+					});
+				return cmd;
+			})
+			.addSubcommand(cmd =>{
+				cmd.setName('static')
+					.setDescription('Enable static slow mode')
+					.addNumberOption(option => {
+						option.setName('limit')
+							.setDescription('The static limit to set the channel slow mode to')
+							.setMinValue(1)
+							.setRequired(true);
+						return option;
+					})
+					.addChannelOption(option => {
+						option.setName('channel')
+							.setDescription('The channel to enable static slow mode for');
+						return option;
+					});
+				return cmd;
+			});
+		return group;
+	})
+	.addSubcommand(cmd => {
+		cmd.setName('disable')
+			.setDescription('Disable slow mode for a channel')
+			.addChannelOption(option => {
+				option.setName('channel')
+					.setDescription('The channel to disable auto slow mode for');
+				return option;
+			});
+		return cmd;
+	})
+	.addSubcommand(cmd => {
+		cmd.setName('stats')
+			.setDescription('Get the current auto slow mode stats for a channel')
+			.addChannelOption(option => {
+				option.setName('channel')
+					.setDescription('The channel to fetch the auto slow mode stats for');
+				return option;
+			});
+		return cmd;
+	})
+	.addSubcommandGroup(group => {
+		group.setName('stage')
+			.setDescription('Managed the auto slow mode threshold settings')
+			.addSubcommand(cmd => {
+				cmd.setName('set')
+					.setDescription('Set the properties of a stage for the given channel')
+					.addNumberOption(option => {
+						option.setName('threshold')
+							.setDescription('the threshold in messages per section that must be reached to enable this stage')
+							.setRequired(true)
+							.setMinValue(0);
+						return option;
+					})
+					.addNumberOption(option => {
+						option.setName('limit')
+							.setDescription('the limit to apply to the channel once the threshold has been reached')
+							.setRequired(true)
+							.setMinValue(0);
+						return option;
+					})
+					.addChannelOption(option => {
+						option.setName('channel')
+							.setDescription('The channel to add a threshold for');
+						return option;
+					});
+				return cmd;
+			})
+			.addSubcommand(cmd => {
+				cmd.setName('unset')
+					.setDescription('Unset a stage for the given channel')
+					.addNumberOption(option => {
+						option.setName('threshold')
+							.setDescription('The index of the threshold to remove')
```

Comment:
Minor wording nitpick, it's not the index of the threshold but the value of the threshold.